api/src/services/collections.ts

```diff
@@ -145,7 +145,24 @@ export class CollectionsService {
 
 					const fieldPayloads = payload.fields!.filter((field) => field.meta).map((field) => field.meta) as FieldMeta[];
 
-					await fieldItemsService.createMany(fieldPayloads, {
+					let sortedFieldPayloads = fieldPayloads
+						.filter((field) => field?.group === undefined || field?.group === null)
+						.map((field, index) => merge({ sort: index + 1 }, field));
+
+					if (sortedFieldPayloads.length < fieldPayloads.length) {
+						const fieldsWithGroups = groupBy(
+							fieldPayloads.filter((field) => field?.group),
+							(field) => field?.group
+						);
+
+						for (const [_group, fields] of Object.entries(fieldsWithGroups)) {
+							sortedFieldPayloads = sortedFieldPayloads.concat(
+								fields.map((field, index) => merge({ sort: index + 1 }, field))
+							);
+						}
+					}
+
+					await fieldItemsService.createMany(sortedFieldPayloads, {
```

Comment:
this logic looks actually complicated enough for me to extract it to a separate function and write unit tests for it 🤔 what do you think @azrikahar @rijkvanzanten

Reply:
Yeah, I think a unit test would be appropriate here 👍

Reply:
as discussed, no unit tests in this PR

Reply:
> as discussed, no unit tests in this PR

In place of a unit test here, @azrikahar would you be able to add a few explaining comments to make sure we'll understand the logic at a later point?

Reply:
> In place of a unit test here, @azrikahar would you be able to add a few explaining comments to make sure we'll understand the logic at a later point?

@paescuj added comments to clarify the logic used here 👍

Reply:
Perfect! ❤️